TELCODOCS-899 Doc customer feedback on rootDeviceHints (#65708)
openshift/openshift-docs: modules/ipi-install-troubleshooting-misc-issues.adoc

+
On high availability clusters with three or more control plane (master) nodes, the Operator will perform leader election and all other Operators will sleep. For additional details, see https://github.com/openshift/installer/blob/master/docs/user/troubleshooting.md[Troubleshooting].

> == Addressing the `No disk found with matching rootDeviceHints` error

Reviewer: ```suggestion
== Addressing the "No disk found with matching rootDeviceHints" error message
```
- Per [OCP repo guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#system-messages), `Always refer to a message with the type of message it is, followed by the word "message". For example, refer to an error message as an "error message", and not simply as an "error".`
- [No back ticks allowed in headings](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#object-references).

After you deploy a cluster, you might receive the following error:

----
`No disk found with matching rootDeviceHints`
----

To address the "No disk found with matching rootDeviceHints" error, a temporary workaround is to change the `rootDeviceHints` to `minSizeGigabytes: 300`.

> After you change the `rootDeviceHints`, boot the CoreOS and then verify the disk information by using the following command:

Reviewer: ```suggestion
After you change the `rootDeviceHints` settings, boot the CoreOS and then verify the disk information by using the following command:
```
Per OCP repo guidelines, always include the type of object after the back-ticked object name: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#object-references

+
On high availability clusters with three or more control plane (master) nodes, the Operator will perform leader election and all other Operators will sleep. For additional details, see https://github.com/openshift/installer/blob/master/docs/user/troubleshooting.md[Troubleshooting].

== Addressing the `No disk found with matching rootDeviceHints` error

> After the deployment of a cluster you might receive the following error:

Reviewer 2: ```suggestion
After you deploy a cluster, you might receive the following error:
```
[IBM Style](https://www.ibm.com/docs/en/ibm-style?topic=grammar-verbs#general-guidelines): Avoid dangling modifiers.

----
`No disk found with matching rootDeviceHints`
----

To address the "No disk found with matching rootDeviceHints" error, a temporary workaround is to change the `rootDeviceHints` to `minSizeGigabytes: 300`.

> After changing the `rootDeviceHints`, boot the CoreOS and then verify the disk information by using the command:

Reviewer 2: ```suggestion
After you change the `rootDeviceHints`, start the CoreOS and then verify the disk information by using the following command:
```
- IBM Style recommends "start" instead of "boot" where possible.
- By including the word "you", you can better engage the reader.

> [source]

Reviewer 2: ```suggestion
[source,terminal]
```

----
$ udevadm info /dev/sda
----

> If you are using DL360 Gen 10 servers, be aware that they have an SD-card slot which might be assigned the `/dev/sda` device name. If no SD card is present in the server, it can cause conflicts. Ensure that the SD card slot is disabled in the server's BIOS settings.

Reviewer 2: ```suggestion
If you are using DL360 Gen 10 servers, be aware that they have an SD-card slot that might be assigned the `/dev/sda` device name. If no SD card is present in the server, it can cause conflicts. Ensure that the SD card slot is disabled in the server's BIOS settings.
```
Use "that" instead of "which" for restrictive clauses. Reference: https://www.ibm.com/docs/en/ibm-style?topic=punctuation-commas#commas-with-nonrestrictive-clauses-and-phrases

> If the `minSizeGigabytes` workaround is not fulfilling the requirements, you might need to revert `rootDeviceHints` back to `/dev/sda`. This change allows ironic images to boot successfully.

Reviewer 2: ```suggestion
If the `minSizeGigabytes` workaround is not fulfilling the requirements, you might need to revert `rootDeviceHints` back to `/dev/sda`. This change allows ironic images to start successfully.
```

Reviewer 2: Ok, no objections here.

> An alternative approach to fixing this problem is by using the serial ID of the disk. However, be aware that finding the serial ID can be challenging and might make the configuration file less readable. If you choose this path, ensure that you gather the serial ID correctly using the command `udevadm info /dev/sda` and incorporate it into your configuration.

Reviewer 2: ```suggestion
An alternative approach to fixing this problem is by using the serial ID of the disk. However, be aware that finding the serial ID can be challenging and might make the configuration file less readable. If you choose this path, ensure that you gather the serial ID correctly by using the following command and incorporating it into your configuration:

[source,terminal]
----
$ udevadm info /dev/sda
----
```
Per the OCP doc guidelines, I believe commands should be separated out on their own lines.

> By following these troubleshooting steps, you can address the issue of "No disk found with matching rootDeviceHints" on your control plane nodes in your {product-title} cluster.

Reviewer 2: In the spirit of minimalism, you can consider removing line 103. Its meaning is already implied by the rest of the text in this troubleshooting entry.
